File: aoc-2023/src/test/java/com/capital7software/aoc/aoc2023/days/Day08Test.java
package com.capital7software.aoc.aoc2023.days;

import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.logging.Logger;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day08Test extends AdventOfCodeTestBase {
    private static final Logger LOGGER = Logger.getLogger(Day08Test.class.getName());

    @BeforeEach
    void setUp() {
        var instance = new Day08();
        setupFromFile(instance.getDefaultInputFilename());
    }

    @Test
    void calculateStepsToTraverseWastelandSingleNodeLow() {
        var instance = new Day08();
        int expected = 2;
        var actual = instance.calculateStepsToTraverseWastelandSingleNode(lines);
        assertEquals(expected, actual, "The total steps taken " +
                "is not what was expected: " + expected);
    }

    @Test
    void calculateStepsToTraverseWastelandSingleNodeHigh() {
        var instance = new Day08();
        setupFromFile("inputs/input_day_08-02.txt");
        int expected = 6;
        var actual = instance.calculateStepsToTraverseWastelandSingleNode(lines);
        assertEquals(expected, actual, "The total steps taken " +
                "is not what was expected: " + expected);
    }

    @Test
    void calculateStepsToTraverseWastelandMultipleNodes() {
        var instance = new Day08();
        setupFromFile("inputs/input_day_08-03.txt");
        int expected = 6;
        var actual = instance.calculateStepsToTraverseWastelandMultipleNodes(lines);
        assertEquals(expected, actual, "The total winnings using Jokers " +
                "is not what was expected: " + expected);
    }

    @Override
    protected Logger getLogger() {
        return LOGGER;
    }
}

File: aoc-2016/src/test/kotlin/com/capital7software/aoc/aoc2016aoc/days/Day08Test.kt
package com.capital7software.aoc.aoc2016aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day08Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day08Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day08()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun testNumberOfPixelsThatShouldBeLit() {
    val instance = Day08()
    val expected = 6L
    val actual = instance.numberOfPixelsThatShouldBeLit(lines)

    assertEquals(expected, actual, "The number of pixels that should be lit "
        + "is not what was expected: $expected")
  }

  @Test
  fun testGetDisplayMessage() {
    val instance = Day08()
    val expected = """LightGrid - Columns: 50, Rows: 6, Instructions: 4
    1 1                                           
1 1                                               
 1                                                
 1                                                
                                                  
                                                  
"""
    val actual = instance.getDisplayMessage(lines)

    assertEquals(expected, actual, "The display message "
        + "is not what was expected: $expected")
  }
}
